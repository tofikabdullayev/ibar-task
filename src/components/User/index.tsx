import React from 'react';
import { ListGroupItem } from 'reactstrap';
import { Link } from 'react-router-dom';
import { User } from '../../store/interfaces';

interface Props extends User {}

const UserItem = ({ first_name, last_name, id }: Props) => {
  const fullName = `${first_name} ${last_name}`;
  return (
    <Link to={`/user/${id}`}>
      <ListGroupItem>{fullName}</ListGroupItem>
    </Link>
  );
};

export default UserItem;
