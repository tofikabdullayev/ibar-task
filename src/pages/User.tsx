import React, { useEffect } from 'react';
import { RouteComponentProps, Link } from 'react-router-dom';
import { Button, ButtonGroup, Spinner } from 'reactstrap';
import { useDispatch, useSelector } from 'react-redux';
import { getUser } from '../store/actions/user';
import { UserState } from '../store/reducers/user';

interface RouteParams {
  userId: string;
}

interface Props extends RouteComponentProps<RouteParams> {}

const data = {
  _meta: {
    success: true,
    code: 200,
    message: 'OK. Everything worked as expected.',
    rateLimit: { limit: 30, remaining: 29, reset: 2 },
  },
  result: {
    id: '11918',
    first_name: 'Emilia',
    last_name: 'Robel',
    gender: 'female',
    dob: '1949-01-14',
    email: 'ryan.agnes@example.com',
    phone: '(514) 618-8655 x26218',
    website: 'http://www.volkman.com/nisi-amet-unde-numquam-id-asperiores',
    address: '149 Weissnat Forks\nSouth Toreybury, ME 50711-7903',
    status: 'inactive',
    _links: {
      self: { href: 'https://gorest.co.in/public-api/users/11918' },
      edit: { href: 'https://gorest.co.in/public-api/users/11918' },
      avatar: { href: 'https://lorempixel.com/250/250/people/?29241' },
    },
  },
};

const User = (props: Props) => {
  const userId = props.match.params.userId;
  const dispatch = useDispatch();
  const userState = useSelector(
    (state: { selectedUser: UserState }) => state.selectedUser
  );

  useEffect(() => {
    dispatch(getUser(userId));
  }, [dispatch, userId]);

  return (
    <div>
      <h2>User Info</h2>
      {userState.isLoading ? (
        <Spinner
          color="primary"
          style={{ display: 'block', margin: '0 auto' }}
        />
      ) : (
        <>
          <p>
            Name:{' '}
            <strong>
              {userState.user.result.first_name +
                ' ' +
                userState.user.result.last_name}
            </strong>
          </p>
          <p>
            Email: <strong>{userState.user.result.email}</strong>
          </p>
          <ButtonGroup>
            <Button color="success">Edit</Button>
            <Button color="danger" onClick={() => alert('removed!')}>
              Remove
            </Button>
            <Button color="primary">
              <Link
                to={`/user/${userId}/posts`}
                style={{ color: 'inherit', textDecoration: 'inherit' }}
              >
                Posts
              </Link>
            </Button>
          </ButtonGroup>
        </>
      )}
    </div>
  );
};

export default User;
