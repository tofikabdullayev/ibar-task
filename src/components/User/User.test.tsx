import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import User from './index';

configure({ adapter: new Adapter() });
test('renders the component', () => {
  const component = shallow(<User first_name="" last_name="" id="" />);
  expect(component).toMatchSnapshot();
});
