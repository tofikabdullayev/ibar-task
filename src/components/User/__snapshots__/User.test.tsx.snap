// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Link
  to="/user/"
>
  <ListGroupItem
    tag="li"
  >
     
  </ListGroupItem>
</Link>
`;
